src/utils/wallet.js

```diff
@@ -445,7 +445,10 @@ class Wallet {
         // TODO: getXXX methods shouldn't be modifying the state
         await setKeyMeta(publicKey, { type: 'ledger' })
 
-        const accountIds = await getAccountIds(publicKey.toString())
+        // TODO: switch to indexer when working properly for keys added / deleted by contracts (e.g. linkdrop)
```

Comment:
no need to switch actually, in both cases you need to call contract helper, switch can happen inside